### PR TITLE
Wrapper for Groups API + tests

### PR DESCRIPTION
**Pull request by aguynamedloren**
- Wraps TrueVault Groups API endpoints: https://docs.truevault.com/groups
- Follows pattern of existing wrappers
- Endpoints: `Create`, `Read`, `List`, `Update`, `Delete`
- Additionally, defined `add_users` and `remove_users` for convenience, which use the `Update` endpoint

Something that bothers me about the tests.. the way VCR cassettes are used here kind of defeats the purpose of testing.  After the first run, the responses are cached regardless of what is sent on subsequent test runs.. meaning one could break the API wrapper and tests would still pass.

### CODE REVIEW

**Comment by aguynamedloren** [comment]
I haven't dug into this VCR concern.. but I think it may be the specific implementation that's troubling.  Requests are matched on method (`:match_requests_on => [:method]`) so parameters are ignored.. Perhaps we should take params into consideration.

File: lib/truevault/group.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      query[:name] = options[:name] if options[:name]
+      query[:policy] = policy(options) if options[:policy]
+      query[:user_ids] = to_string(options[:user_ids]) if options[:user_ids]
+      query[:operation] = options[:operation] if options[:operation]
+
+      new_options = default_options_to_merge_with.merge(query: query)
+      self.class.put("/#{@api_ver}/groups/#{group_id}", new_options)
+    end
+
+    def add_users(group_id, user_ids, options = {})
```
Comment (drewbaumann) [inline lib/truevault/group.rb]: Is options getting used here?

File: lib/truevault/group.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        query: {
+          user_ids: to_string(user_ids),
+          operation: "APPEND"
+          }
+        }
+      new_options = default_options_to_merge_with.merge(query)
+      self.class.put("/#{@api_ver}/groups/#{group_id}", new_options)
+    end
+
+    def remove_users(group_id, user_ids, options = {})
```
Comment (drewbaumann) [inline lib/truevault/group.rb]: same question regarding options here

File: lib/truevault/group.rb (hunk 1 above)
Comment (aguynamedloren) [inline lib/truevault/group.rb]: good call, nope

File: lib/truevault/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module TrueVault
-  VERSION = "0.0.4"
+  VERSION = "0.1.0"
```
Comment (drewbaumann) [inline lib/truevault/version.rb:2]: Woah. 1.0 eh?

**Comment by drewbaumann** [comment]
other than the options this LGTM. Dont really feel urgency around the tests unless you're planning on submitting this PR to the real gem.

File: lib/truevault/version.rb (hunk 3 above)
Comment (aguynamedloren) [inline lib/truevault/version.rb:2]: no, it's `0.1.0` lol minor version bump

**Comment by aguynamedloren** [comment]
@drewbaumann yeah, I'm thinking this is probably a good candidate for the TrueVault gem!

**Comment by aguynamedloren** [comment]
@drewbaumann updated!

**Comment by drewbaumann** [comment]
Great. Can you merge this or do I have to?

**Comment by aguynamedloren** [comment]
screen shot 2016-07-21 at 2 33 07 pm

looks like you have to

**Comment by drewbaumann** [comment]
@aguynamedloren merged

**Pull request merged**